vms/avm/vm_test.go

```diff
@@ -606,6 +623,74 @@ func TestGenesisGetUTXOs(t *testing.T) {
 	}
 }
 
+// TestGenesisGetPaginatedUTXOs tests
+// - Pagination when the total UTXOs exceed maxUTXOsToFetch (1024)
+// - Fetching all UTXOs when they exceed maxUTXOsToFetch (1024)
+func TestGenesisGetPaginatedUTXOs(t *testing.T) {
+
+	addr0Str, _ := formatting.FormatBech32(testHRP, addrs[0].Bytes())
+
+	// Create a starting point of 2000 UTXOs
+	utxoCount := 2000
+	holder := map[string][]interface{}{}
+	for i := 0; i < utxoCount; i++ {
+		holder["fixedCap"] = append(holder["fixedCap"], Holder{
+			Amount:  json.Uint64(startBalance),
+			Address: addr0Str,
+		})
+	}
+
+	// Inject them in the Genesis build
+	genesisArgs := &BuildGenesisArgs{GenesisData: map[string]AssetDefinition{
+		"asset1": {
+			Name:         "AVAX",
+			Symbol:       "SYMB",
+			InitialState: holder,
+		},
+	}}
+	_, _, vm, _ := GenesisVMWithArgs(t, genesisArgs)
+	ctx := vm.ctx
+	defer func() {
+		if err := vm.Shutdown(); err != nil {
+			t.Fatal(err)
+		}
+		ctx.Lock.Unlock()
+	}()
+
+	addrsSet := ids.ShortSet{}
+	addrsSet.Add(addrs[0])
+
+	// First Page - using paginated calls
+	paginatedUTXOs, lastAddr, lastIdx, err := vm.GetUTXOs(addrsSet, ids.ShortEmpty, ids.Empty, -1, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(paginatedUTXOs) == utxoCount {
+		t.Fatalf("Wrong number of utxos. Should be Paginated. Expected (%d) returned (%d)", maxUTXOsToFetch, len(paginatedUTXOs))
+	}
+
+	// Last Page - using paginated calls
+	paginatedUTXOsLastPage, _, _, err := vm.GetUTXOs(addrsSet, lastAddr, lastIdx, -1, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(paginatedUTXOs)+len(paginatedUTXOsLastPage) != utxoCount {
+		t.Fatalf("Wrong number of utxos. Should have paginated through all. Expected (%d) returned (%d)", utxoCount, len(paginatedUTXOs)+len(paginatedUTXOsLastPage))
+	}
```

Comment:
I think this is check is invalid if UTXOs reference multiple addresses. Which is intended behavior. Nothing to change based on this comment, just to make sure you're aware.

Reply:
Yeah I get it 👍 

We're just using one address with a bunch of utxos - potentially we could test multiple addresses too.